build-system/pr-check.js

```diff
@@ -91,37 +102,106 @@ function isValidatorFile(filePath) {
   if (filePaths.length == 0) {
-    return new Set(['VALIDATOR_WEBUI', 'VALIDATOR', 'RUNTIME']);
+    return new Set(['BUILD_SYSTEM', 'VALIDATOR_WEBUI', 'VALIDATOR', 'RUNTIME',
+        'DOCS']);
   }
   const targetSet = new Set();
   for (p of filePaths) {
-    if (isValidatorWebuiFile(p)) {
+    if (isBuildSystemFile(p)) {
+      targetSet.add('BUILD_SYSTEM');
+    } else if (isValidatorWebuiFile(p)) {
       targetSet.add('VALIDATOR_WEBUI');
     } else if (isValidatorFile(p)) {
       targetSet.add('VALIDATOR');
+    } else if (isDocFile(p)) {
+      targetSet.add('DOCS');
     } else {
       targetSet.add('RUNTIME');
     }
   }
   return targetSet;
 }
 
+
+const command = {
+  testBuildSystem: function() {
+    execOrDie('npm run ava');
+  },
+  buildRuntime: function() {
+    execOrDie(`${gulp} lint`);
+    execOrDie(`${gulp} build --css-only`);
+    execOrDie(`${gulp} check-types`);
+    execOrDie(`${gulp} dist --fortesting`);
+  },
+  testRuntime: function() {
+    // dep-check needs to occur after build since we rely on build to generate
+    // the css files into js files.
+    execOrDie(`${gulp} dep-check`);
+    // Unit tests with Travis' default chromium
+    execOrDie(`${gulp} test --nobuild --compiled`);
+    // Integration tests with all saucelabs browsers
+    execOrDie(`${gulp} test --nobuild --saucelabs --integration --compiled`);
+    // All unit tests with an old chrome (best we can do right now to pass tests
+    // and not start relying on new features).
+    // Disabled because it regressed. Better to run the other saucelabs tests.
+    execOrDie(`${gulp} test --nobuild --saucelabs --oldchrome --compiled`);
+  },
+  presubmit: function() {
+    execOrDie(`${gulp} presubmit`);
+  },
+  buildValidatorWebUI: function() {
+    execOrDie('cd validator/webui && python build.py');
+  },
+  buildValidator: function() {
+    execOrDie('cd validator && python build.py');
+  },
+};
+
+function runAllCommands() {
+  command.testBuildSystem();
+  command.buildRuntime();
+  command.presubmit();
+  command.testRuntime();
+  command.buildValidatorWebUI();
+  command.buildValidator();
+}
+
 /**
  * The main method for the script execution which much like a C main function
  * receives the command line arguments and returns an exit status.
  * @param {!Array<string>} argv
  * @returns {number}
  */
 function main(argv) {
-  const travisCommitRange = argv[2] || '';
-  const buildTargets = determineBuildTargets(filesInPr(travisCommitRange));
+  // If $TRAVIS_PULL_REQUEST_SHA is empty then it is a push build and not a PR.
+  if (!process.env.TRAVIS_PULL_REQUEST_SHA) {
+    console.log('Running all commands on push build.');
+    runAllCommands();
+    return 0;
+  }
+  const travisCommitRange = `master...${process.env.TRAVIS_PULL_REQUEST_SHA}`;
+  const files = filesInPr(travisCommitRange);
+  const buildTargets = determineBuildTargets(files);
+
+  if (buildTargets.length == 1 && buildTargets[0] == 'DOCS') {
```

Comment:
`buildTargets` is a `Set`, so this will never be true. `buildTargets.has('DOCS')`